src/end_entity.rs

```diff
@@ -77,16 +77,16 @@ impl<'a> EndEntityCert<'a> {
     }
 
     #[allow(clippy::too_many_arguments)]
-    fn verify_is_valid_cert(
+    fn verify_is_valid_cert<'b, 'c>(
         &self,
-        supported_sig_algs: &[&SignatureAlgorithm],
-        trust_anchors: &[TrustAnchor],
-        intermediate_certs: &[&[u8]],
+        supported_sig_algs: &'c [&SignatureAlgorithm],
+        trust_anchors: &'b [TrustAnchor<'b>],
```

Comment:
I think we can assume `'c` is never longer than `'b`, so replacing this with `&'c [Trustanchor<'b>]` may improve usability.